\name{readFile}
\alias{readFile}
%- Also NEED an '\alias' for EACH other topic documented here.
\title{
This is a function read in the intermediated file generated by fromat_grand.p
}
\description{
%%  ~~ A concise (1-5 lines) description of what the function does. ~~
}
\usage{
readFile(x)
}
%- maybe also 'usage' for other objects documented here.
\arguments{
  \item{pathFile}{
%%     ~~Describe \code{x} here~~
the file path of a intermediate file
}
}
\details{
%%  ~~ If necessary, more details than the description above ~~
}
\value{
%%  ~Describe the value returned
%%  If it is a LIST, use
%%  \item{comp1 }{Description of 'comp1'}
%%  \item{comp2 }{Description of 'comp2'}
%% ...
return a data.table object with gentoype of the divergent lines and this indiviudal in columns. Each row is a site which is fixed between parental and has genotype call for this particular offspring.

}
\references{
Yanjun Zan, Thibaut Payen, Leif Andersson, Paul B. Siegel and Ã–rjan Carlborg;Whole-genome QTL mapping in experimental pedigrees from outbred founders utilizing low coverage individual based sequencing 2018
}
\author{
Yanjun Zan, Thibaut Payen
}
\note{
%%  ~~further notes~~
}

%% ~Make other sections like Warning with \section{Warning }{....} ~

\seealso{
%% ~~objects to See Also as \code{\link{help}}, ~~~
}
\examples{
##---- Should be DIRECTLY executable !! ----
##-- ==>  Define data, use random,
##--	or do  help(data=index)  for the standard data sets.

## The function is currently defined as
}
% Add one or more standard keywords, see file 'KEYWORDS' in the
% R documentation directory.
\keyword{ ~kwd1 }
\keyword{ ~kwd2 }% __ONLY ONE__ keyword per line
